pkg/hubble/relay/server/server.go

```diff
@@ -90,12 +90,8 @@ func New(options ...Option) (*Server, error) {
 		pool.WithPeerClientBuilder(peerClientBuilder),
 		pool.WithClientConnBuilder(pool.GRPCClientConnBuilder{
 			DialTimeout: opts.dialTimeout,
```

Comment:
With all the unused `DialTimeout`s I think we can then actually get rid of the `dialTimeout` option. Which would then even mean that we can get rid of the the `--dial-timeout` flag for relay